crates/doc/src/extractor.rs

```diff
@@ -93,12 +121,41 @@ impl Extractor {
         doc: &N,
```

Comment:
drive by nit on your WIP: how about not changing the interface for `extract_all`, and having a new function `extract_all_track_pruned` that takes a &AtomicBool (non-optional), and then have `extract_all` delegate to it with a stack-local bool? `sentinel` -> `pruned` (i'm not sure what sentinel means in this context)?

There are many _other_ places we do extraction where we simply don't care about this pruning bit.